### ISSUE TITLE
[Bug]: Track order is broken when sorting by album in playlists

### ISSUE THREAD

**Issue by lilacpixel** [navidrome/navidrome#3177]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.52.5

### Current Behavior

When sorting playlists by album, tracks within albums may be placed into illogical groupings, with no respect for the original track order. For example, a two-disc soundtrack shows track 11 of disc 2 at the top of the tracks for that particular album. The tracks appear in the correct order on the album itself. Other software that makes use of the Subsonic API, such as Ampache, displays the same albums with the correct track order within playlists.

### Expected Behavior

Album-based sorting should sort tracks properly based on track and disc order.

### Steps To Reproduce

1. Create a smart playlist with album sort and ascending order.
2. Check the ordering for any set of tracks within an album.
3. Observe that tracks appear out of order.

### Environment

```markdown
- OS: Debian Bookworm
- Browser: Firefox 128.0.3
- Client: Feishin 0.7.3
```

### How Navidrome is installed?

Docker

### Anything else?

Correct track order within the album view:
![Screenshot 2024-07-30 at 4 57 01 PM](https://github.com/user-attachments/assets/7283cda9-9cc9-458b-b5c5-24f22c3b72b5)

Partial track order within a playlist sorted by album:
![Screenshot 2024-07-30 at 4 57 28 PM](https://github.com/user-attachments/assets/fef28e6d-ab2a-4d93-be81-807498c96795)

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by certuna** [comment]
Yes, I think this is a bug in the WebUI, it doesn't sort by disc number. I think we can tackle this best after the UI migration to ReactAdmin v3 -> v5?

**Comment by Szeraax** [comment]
Great, then I'll start tracking this for that milestone.

**Comment by lilacpixel** [comment]
Replying to keep open--bug is still present in the latest version. Thanks!

**Comment by lilacpixel** [comment]
Still an issue--thanks!

**Comment by deluan** [comment]
Hey @lilacpixel, sorry for missing this. Can you try PR #4657?

**Comment by lilacpixel** [comment]
> Hey [@lilacpixel](https://github.com/lilacpixel), sorry for missing this. Can you try PR [#4657](https://github.com/navidrome/navidrome/pull/4657)?

No worries; thanks so much for taking another look at this!

The track order when using `deluan/navidrome:pr-4657`, weirdly, is different, but still wrong.

<img width="1548" height="398" alt="Image" src="https://github.com/user-attachments/assets/a9f98fb9-0132-4e13-a775-0ff5a309795a" />

At a glance, it looks like it's no longer mixing up tracks across multiple discs, but within disc 1, for instance, the track order goes 1, then 9, then 6, then 8, etc.

I tried running a full scan to see if that helped, but no such luck. Is there anything else I might need to do to ensure the fix applies properly?

Just in case it helps, this is the playlist I'm looking at:

```
{
  "name": "User's Library",
  "all": [
  	{"is": {"library_id": 3}}
  ],
  "sort": "album",
  "order": "asc"
}
```

Thanks again for your work on this, I appreciate it! 😸

**Comment by deluan** [comment]
Ah! Sorry, I missed the part where you mentioned Smart Playlists. The fix in the PR is for the UI. For Smart Playlists it will require some more work. A workaround is to specify all fields that you want to sort by:
```
{
  "name": "User's Library",
  "all": [
  	{"is": {"library_id": 3}}
  ],
  "sort": "album,discnumber,tracknumber",
  "order": "asc"
}
```

I'll keep this open as I'm planning to rework the Smart Playlists implementation and will support this better.

**Comment by lilacpixel** [comment]
Thanks for sharing a workaround! For whatever reason, it doesn't seem to be changing the sort order for me, but it's also far from a critical feature. I'll keep an eye out for a future update to Smart Playlists! 😊

**Comment by deluan** [comment]
> Thanks for sharing a workaround! For whatever reason, it doesn't seem to be changing the sort order for me, but it's also far from a critical feature. I'll keep an eye out for a future update to Smart Playlists! 😊

```bash
  "sort": "album,discnumber,tracknumber",
```

This was introduced in recent versions, are you still using 0.52.5? If so, you really should upgrade.

**Comment by lilacpixel** [comment]
> This was introduced in recent versions, are you still using 0.52.5? If so, you really should upgrade.

I'm on 0.58.0 (`9dbe0c18` according to the about modal in the app). With the following playlist, the track order is the same as in the above screenshot:

```
{
  "name": "User's Library",
  "all": [
  	{"is": {"library_id": 3}}
  ],
  "sort": "album,discnumber,tracknumber",
  "order": "asc"
}
```

Is there anything else I'm missing? Happy to open a new issue, if this does end up being a different bug!

**Comment by deluan** [comment]
Not sure what is going on. I tried this same NSP with my library and it sorts the tracks as expected.

**Comment by lilacpixel** [comment]
I was able to get the intended sort order with `"sort": "album,discnumber,tracknumber"` only by performing an edit operation in the playlist settings. In my case, changing the playlist owner was enough to force Navidrome to rescan the playlist, after which I was able to change it back.

I would expect changes to a Smart Playlist to get picked up on the next library scan, so this feels like a bug to me. I didn't see anything that seemed related in open or closed issues--happy to open a separate issue if this is something that might be actionable ahead of the Smart Playlists rework.

Thanks again!

**Comment by deluan** [comment]
Before opening a new issue, make sure to reproduce it. I tried it locally and Navidrome picks up changes to .nsp files on the next scan. 

If you can reproduce it, then go ahead and open an issue, but add as much details as possible, including (debug) logs of the scan after the nsp change.
